src/cloud-element-templates/behavior/ConditionalBehavior.js

```diff
@@ -80,6 +80,10 @@ export default class ConditionalBehavior extends CommandInterceptor {
       return;
     }
 
+    if (template.id !== oldTemplate.id) {
```

Comment:
What feature does this implement? Could we add a dedicated test case to verify it somehow? Does this break existing behavior?

I could, for example imagine that we migrate from one template to another (different ID), but still keep most properties ("migration after deprecating the old one"). 

In that case, are conditional bindings in the new template still applied, I guess so?

Reply:
This ensures that we do not apply conditions (and change the element type) when we change the element template ID in the moodle here: 
https://github.com/bpmn-io/bpmn-js-element-templates/blob/921634c208387c2bf2f8c37fe05683a90199315f/src/element-templates/cmd/ChangeElementTemplateHandler.js#L78

This prevents:
```
ChangeElementTemplateHandler
├── change `camunda:modelerTemplate
│   └── `updateProperties` triggers conditional logic
│       └── ChangeElementTemplateHandler
│           ├── ...
│           └── replace element with new type (old element gets removed)
├── ...
└── replace (already removed) element with new type ==> error
```

We still apply conditions to the new template in the `propertiesPanel.zeebe.changeTemplate` hook, just not in the `updateProperties` hook that updates the ID in the XML. I will adjust the code to better reflect that

Reply:
I made the check more specific in [4e0c2cc](https://github.com/bpmn-io/bpmn-js-element-templates/pull/111/commits/4e0c2cc90125145beb49f1a423f584373d6dee83). 

Due to the complex nature of this bug, I find it hard to create a specific test case for it. The integration test `Github message start => Webhook blank-start` will fail without it :)